src/cr/cube/dimension.py

```diff
@@ -57,6 +57,17 @@ def apparent_dimensions(self):
         """
         return _ApparentDimensions(all_dimensions=self._dimensions)
 
+    @lazyproperty
+    def require_transposition(self):
+        """True if dimensions need transposition, False otherwise.
+
+        Transposition is needed when one of the dimension type is a NUM_ARRAY and the
+        dimensions are more than 1. This business rule needs to drive correctly tabbook
+        and deck exports when a numeric array dimension is expressed.
+        """
+        dimension_types = [d.dimension_type for d in self.apparent_dimensions]
+        return len(self.apparent_dimensions) >= 2 and DT.NUM_ARRAY in dimension_types
```

Comment:
If this is a temporary hack while we wait for `dim_order`, I'm a little uncomfortable with allowing the hack to spread to the `cr.cube.dimension` module where we're likely to forget to remove it when the time comes. I think this would be better appearing only in `cr.cube.cube.Cube` and marked as a hack and when it is expected to come out.

Reply:
Considering that this property is needed even in `_BaseMeasure` in the shape, i'd prefer to have it in something where is effectively related. We are transposing the dimensions so... dimensions would be an appropriate module.